From 265d1ea3407e072e635c6d3b91a17fad41920b8c Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Tue, 9 May 2017 10:49:10 +0800
Subject: [PATCH 1770/5242] MLK-14807-2: ARM: dts: imx6: correct all
 ldo-enable dts

commit  63d1a697d22e7b0ff546e1da6380ef640bef9511 from
https://source.codeaurora.org/external/imx/linux-imx.git

'vin-supply' property used in ldo-bypass mode while 'vin-supply' deleted
in ldo-enable mode on v4.9 rather than dirctly switch 'supply' to internal
regulator and external pmic regulator on v4.1. Correct it for all *-ldo.dts,
otherwise, still work in ldo-bypass mode.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-ldo.dts |   10 ++++++++--
 arch/arm/boot/dts/imx6ull-9x9-evk-ldo.dts         |   10 ++++++++--
 2 files changed, 16 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-ldo.dts b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-ldo.dts
index 735b2ef..4266dc24 100644
--- a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-ldo.dts
+++ b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-ldo.dts
@@ -25,11 +25,17 @@
 		396000	1175000
 		198000	1175000
 	>;
-	arm-supply = <&reg_arm>;
-	soc-supply = <&reg_soc>;
 	fsl,arm-soc-shared = <0>;
 };
 
 &gpc {
 	fsl,ldo-bypass = <0>; /* use ldo-enable, u-boot will check it and configure */
 };
+
+&reg_arm {
+	/delete-property/ vin-supply;
+};
+
+&reg_soc {
+	/delete-property/ vin-supply;
+};
diff --git a/arch/arm/boot/dts/imx6ull-9x9-evk-ldo.dts b/arch/arm/boot/dts/imx6ull-9x9-evk-ldo.dts
index 817f48a..a878fe5 100644
--- a/arch/arm/boot/dts/imx6ull-9x9-evk-ldo.dts
+++ b/arch/arm/boot/dts/imx6ull-9x9-evk-ldo.dts
@@ -20,11 +20,17 @@
 		396000	1175000
 		198000	1175000
 	>;
-	arm-supply = <&reg_arm>;
-	soc-supply = <&reg_soc>;
 	fsl,arm-soc-shared = <0>;
 };
 
 &gpc {
 	fsl,ldo-bypass = <0>; /* use ldo-enable, u-boot will check it and configure */
 };
+
+&reg_arm {
+	/delete-property/ vin-supply;
+};
+
+&reg_soc {
+	/delete-property/ vin-supply;
+};
-- 
1.7.9.5

